dali/operators/decoder/audio/audio_decoder_op.h

```diff
@@ -0,0 +1,135 @@
+#include <vector>
+#include "dali/core/static_switch.h"
+#include "dali/operators/decoder/audio/audio_decoder.h"
+#include "dali/operators/decoder/audio/generic_decoder.h"
+#include "dali/pipeline/data/backend.h"
+#include "dali/pipeline/workspace/workspace.h"
+#include "dali/pipeline/operator/operator.h"
+#include "dali/pipeline/workspace/host_workspace.h"
+
+namespace dali {
+
+namespace detail {
+
+const std::string kOutputTypeName = "dtype";  // NOLINT
+const int kNumOutputs = 2;
+
+}  // namespace detail
+
+class AudioDecoderCpu : public Operator<CPUBackend> {
+ private:
+  using Backend = CPUBackend;
+
+ public:
+  explicit inline AudioDecoderCpu(const OpSpec &spec) :
+          Operator<Backend>(spec),
+          output_type_(spec.GetArgument<DALIDataType>(detail::kOutputTypeName)) {}
+
+
+  inline ~AudioDecoderCpu() override = default;
+
+ protected:
+  bool SetupImpl(std::vector<OutputDesc> &output_desc, const workspace_t<Backend> &ws) override {
+    auto &input = ws.template InputRef<Backend>(0);
+    const auto batch_size = input.shape().num_samples();
+
+    for (int i = 0; i < input.shape().num_samples(); i++) {
+      DALI_ENFORCE(input.shape()[i].size() == 1, "Input must be 1D encoded byte data");
+    }
+    DALI_ENFORCE(IsType<uint8_t>(input.type()), "Input must be stored as uint8_t data.");
+    TypeInfo type;
+    TypeInfo type_i32;
+    type_i32.SetType<int>(DALI_INT32);
+    decoders_.resize(batch_size);
+    sample_meta_.resize(batch_size);
+    files_names_.resize(batch_size);
+
+    TYPE_SWITCH(output_type_, type2id, OutputType, (int16_t, int32_t, float), (
+            for (int i=0; i < batch_size; i++) {
+            decoders_[i] = std::make_unique<GenericAudioDecoder<OutputType>>();
+        }
+        type.SetType<OutputType>(output_type_);
+    ), DALI_FAIL("Unsupported output type"))  // NOLINT
+
+    output_desc.resize(detail::kNumOutputs);
+
+    // Currently, metadata is only the sampling rate.
+    // On the event something else would emerge,
+    // this approach should be completely redefined
+    TensorListShape<> shape_rate(batch_size, 1);
+    TensorListShape<> shape_data(batch_size, 2);
+
+    for (int i = 0; i < batch_size; i++) {
+      auto meta = decoders_[i]->Open({reinterpret_cast<const char *>(input[i].raw_mutable_data()),
+                                      input[i].shape().num_elements()});
+      sample_meta_[i] = meta;
+      shape_data.set_tensor_shape(i, {meta.length, meta.channels});
+      shape_rate.set_tensor_shape(i, {1});
+      files_names_[i] = input[i].GetSourceInfo();
+    }
+    output_desc[0] = {shape_data, type};
+    output_desc[1] = {shape_rate, type_i32};
+    return true;
+  }
+
+
+  void RunImpl(workspace_t<Backend> &ws) override {
+    auto &decoded_output = ws.template OutputRef<Backend>(0);
+    auto &sample_rate_output = ws.template OutputRef<Backend>(1);
+    auto &tp = ws.GetThreadPool();
+
+    for (int i = 0; i < decoded_output.shape().num_samples(); i++) {
```

Comment:
```suggestion
    for (int i = 0; i < batch_size; i++) {
```
suggestion store `batch_size` in a variable before the loop

Reply:
Done